src/connectionconfig/connectionDialogWebViewController.ts

```diff
@@ -314,38 +293,129 @@ export class ConnectionDialogWebViewController extends ReactWebViewPanelControll
 						validationMessage: ''
 					};
 				}
-			},
-			{
-				propertyName: 'database',
-				label: 'Database',
-				required: false,
-				type: FormComponentType.Input,
-			},
-			{
-				propertyName: 'trustServerCertificate',
-				label: 'Trust Server Certificate',
-				required: false,
-				type: FormComponentType.Checkbox,
-			},
+			}
+		});
+
+		components.set('profileName', {
+			option: undefined,
+			component: {
+			propertyName: 'profileName',
+			label: 'Profile Name',
+			required: false,
+			type: FormComponentType.Input,
+			}
+		});
+
+		// add missing validation functions for generated components
+		components.get('server')!.component.validate = (value: string) => {
```

Comment:
it seems like this validation logic could be wired up more automatically using the "required" metadata property?  Since there's only a few properties I'm not sure it's worth refactoring though.

Reply:
I definitely something worth looking into as the generic form structure gets refined.  It might need more prodding to correctly handle fields that are "conditionally required", like username, which is required, but only if auth is set to SqlLogin.  Lemme see what I can do.